lib/parsers/package-lock-parser.ts

```diff
@@ -120,8 +120,13 @@ export class PackageLockParser implements LockfileParser {
 
     // number of dependencies including root one
     let treeSize = 1;
-
+    const treeSizeLimit = 6.0e6;
+  
     for (const dep of topLevelDeps) {
+      // tree size limit should be 6 millions.
+      if (treeSize > treeSizeLimit) {
+        throw new Error('Tree size has surpassed current tree size limit.');
```

Comment:
Can we please make this a named error like `OutOfSyncError` so we can catch it upstream and present it nicely?